### PR TITLE
chore: bump Google Cloud DLP from 1.0.0 to 3.12.1 and Apache Beam from 2.30.0 to 2.49.0

### LINKED ISSUE #291: Chore: update python dlp client
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave _+1_ or _me too_ comments; they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If the issue is assigned to a user, that user is claiming responsibility for the issue.

### Secured Data Warehouse Specifications

* any version
* update both example pipeline versions (for example [this line](https://github.com/GoogleCloudPlatform/terraform-google-secured-data-warehouse/blob/main/flex-templates/python/regional_dlp_transform/requirements.txt#L2))

### Terraform execution logs

n/a

### Expected Behavior

Use latest DLP library in python

### Actual Behavior

requirements file is using an older pinned version

### Steps to Reproduce

1.  run existing DLP pipieline for python

### References
* [DLP python version](https://github.com/googleapis/python-dlp/releases/tag/v3.6.0)

* #0000

### PR DESCRIPTION
**Pull request by Samir-Cit**
Fixes #291

> Bump Google Cloud DLP from version 1.0.0 to 3.12.1
> Bump Apache Beam from version 2.30.0 to 2.49.0
> Bump Python from 3.7 to 3.8

### CODE REVIEW

File: flex-templates/python/modules/requirements.txt [hunk 1]
```diff
@@ -1,2 +1,2 @@
 apache-beam==2.30.0
-google-cloud-dlp==1.0.0
+google-cloud-dlp==3.6.2
```
Comment (erlanderlo) [inline flex-templates/python/modules/requirements.txt:2]: are there changes needed in the open Java PR that we should consider wrt DLP library version?

**Comment by erlanderlo** [comment]
Could you also please provide evidence of testing such as affirmation / screenshots / outputs that this version of DLP and pipeline working as expected? I don't believe our testing performs any de-id operation or pipeline (aka "workload") validation yet.  For instance, a comment with "I ran the xyz example and confirmed that data was de-id and re-id as expected" helps.  The goal isn't to create added work or to be detailed, unless you have info handy.

Another option to consider (for a future PR?) is to devise a test using the golang framework that mimics what our pipeline would does with DLP - mainly focus in on DLP crypto ops + template API request we use rather than e2e.  However, confirm with the team since that might cause our automated tests to take too long to be viable for our needs.

**Review by erlanderlo — APPROVED** [review]
thanks! LGTM with comments

File: flex-templates/python/modules/requirements.txt (hunk 1 above)
Comment (Samir-Cit) [reply flex-templates/python/modules/requirements.txt:2]: I'll talk to @vfigueiredo-cit regarding this and I'll let you know!

**Comment by Samir-Cit** [comment]
> Could you also please provide evidence of testing such as affirmation / screenshots / outputs that this version of DLP and pipeline working as expected? I don't believe our testing performs any de-id operation or pipeline (aka "workload") validation yet. For instance, a comment with "I ran the xyz example and confirmed that data was de-id and re-id as expected" helps. The goal isn't to create added work or to be detailed, unless you have info handy.
> 
> Another option to consider (for a future PR?) is to devise a test using the golang framework that mimics what our pipeline would does with DLP - mainly focus in on DLP crypto ops + template API request we use rather than e2e. However, confirm with the team since that might cause our automated tests to take too long to be viable for our needs.

I tested for bigquery and pubsub and it's working fine. I was able to see in bigquery that the date was de-id and re-id as expected.

**Follow some examples:**
_Confidential data:_
```
{
  "city": "CANADA",
  "ein": "266641110", 
  "ico": "% FOUR SEASONS",
  "income_amt": "57725551", 
  "name": "VEBA INC VEBA TR",
  "revenue_amt": "57725551",
  "state": "None",
  "street": "TORONTO ONTARIO M3C 2K8"
}
```
_Non-confidential data:_
```
{
  "ein": "AWWdXDxOKxJ005BZbyGqGaNZalWC0BrRh2M=",
  "name": "AdOdV6RvyIyjwD9cCvexrdGzc6\/1bERiMGckA6QY+6K0",
  "ico": "% FOUR SEASONS",
  "street": "ASnm0YirdAN6ivgs5e2zKVCWOPljMaHCaxGz9xpEeI+4ASlG\/TmEvQ==",
  "city": "CANADA",
  "state": "Zucy",
  "income_amt": 57725551,
  "revenue_amt": 57725551
 }
```
_Pubsub:_
```
{
  "book": "Paradise Lost",
  "author": "lLvwB-3R9S9"
 }
```

**Comment by amandakarina** [comment]
De-id

image

image

Re-id
image

image

**Review by lanreogunmola — APPROVED** [review]

**Pull request merged**